main.tf

```diff
@@ -1,65 +1,87 @@
-module "ipam" {
-  source  = "./modules/ipam"
-  tags    = var.ipam.tags
-  cascade = var.ipam.cascade
-}
+#singular
+resource "aws_vpc_ipam" "this" {
+  count = local.create_vpc_ipam && var.vpc_ipam.ipam != null ? 1 : 0
 
-module "vpc_ipam_pool" {
-  source                            = "./modules/pool"
-  address_family                    = var.pool.address_family
-  allocation_default_netmask_length = var.pool.allocation_default_netmask_length
-  allocation_max_netmask_length     = var.pool.allocation_max_netmask_length
-  allocation_min_netmask_length     = var.pool.allocation_min_netmask_length
-  allocation_resource_tags          = var.pool.allocation_resource_tags
-  auto_import                       = var.pool.auto_import
-  aws_service                       = var.pool.aws_service
-  ipam_scope_id                     = var.pool.ipam_scope_id
-  locale                            = var.pool.locale
-  publicly_advertisable             = var.pool.publicly_advertisable
-  public_ip_source                  = var.pool.public_ip_source
-  source_ipam_pool_id               = var.pool.source_ipam_pool_id
-  tags                              = var.pool.tags
+  description = var.vpc_ipam.ipam.description
+  dynamic "operating_regions" {
+    for_each = var.vpc_ipam.ipam.operating_regions
+    content {
+      region_name = operating_regions.value.region_name
+    }
+  }
+  tags    = var.vpc_ipam.ipam.tags
+  cascade = var.vpc_ipam.ipam.cascade
 }
 
-module "vpc_ipam_pool_cidr" {
-  source                               = "./modules/pool-cidr"
-  cidr                                 = var.pool_cidr.cidr
-  cidr_authorization_context_message   = var.pool_cidr.cidr_authorization_context_message
-  cidr_authorization_context_signature = var.pool_cidr.cidr_authorization_context_signature
-  ipam_pool_id                         = var.pool_cidr.ipam_pool_id
-  netmask_length                       = var.pool_cidr.netmask_length
+# #plural, and nestable to 10 deep
+resource "aws_vpc_ipam_pool" "this" {
+  for_each = { for index, pool in coalesce(var.vpc_ipam.pool, []) : index => pool }
+
+  address_family = each.value.address_family
+  #expects lowercase (need conditional)
```

Comment:
curious what thing expects lowercase? you could do something like `try(lower(each.value.....), null)`